tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -372,6 +372,55 @@ create_disk() {
 	OK "Partitions created"
 }
 
+create_erofs_rootfs_image() {
+	local rootfs="$1"
+	local image="$2"
+	local fs_type="$3"
+	local block_size="$4"
+
+	if [ $block_size -ne 4096 ]; then
+		die "Invalid block size for erofs"
+	fi
+
+	readonly mount_dir=$(mktemp -p ${TMPDIR:-/tmp} -d osbuilder-mount-dir.XXXX)
+
+	info "Copying content from rootfs to root partition"
+	cp -a "${rootfs}"/* "${mount_dir}"
+
+	info "Removing unneeded systemd services and sockets"
+	for u in "${systemd_units[@]}"; do
+		find "${mount_dir}" -type f \( \
+			 -name "${u}.service" -o \
+			 -name "${u}.socket" \) \
+			 -exec rm -f {} \;
+	done
+
+	info "Removing unneeded systemd files"
+	for u in "${systemd_files[@]}"; do
+		find "${mount_dir}" -type f -name "${u}" -exec rm -f {} \;
+	done
+
+	info "Creating empty machine-id to allow systemd to bind-mount it"
+	touch "${mount_dir}/etc/machine-id"
+
+	readonly fsimage="$(mktemp)"
+	mkfs.erofs "${fsimage}" "${mount_dir}"
+	rm -rf "${mount_dir}"
+
+	local img_size="$(stat -c"%s" ${fsimage})"
+	local img_size_mb="$((((${img_size} + 1048576) / 1048576) + 1 + ${rootfs_start}))"
```

Comment:
what is this magic number `1048576` ?

Reply:
since mkfs.erofs generates a target image with rootfs dir directly, so I think it needs to align with 1MB, but I'm not quite sure.